ClientWeb/Evolution/Components/gene.component.ts

```diff
@@ -54,6 +55,16 @@ export class GeneComponent extends BaseGeneComponent implements OnInit, OnDestro
         super();
     }
 
+    manuallyCastTo(source: any): Site[] {
+        //Workaround: when we do deep copy of some item with ObjectHelper.deepCopy(), it stores the structure of tree of properties.
+        //Workaround: BUT it also looses the typing, so the structure is the same, but it's anonimous type. And if original type has
+        //Workaround: some computable property, it's lost in anonimous type. It's exactly what happens with Site.name computable prop.
+        //Done: So, to fix this problem created interface ManuallyCast which contains this method manuallyCast and the solution is
+        //Done: RE-CREATING original type by using data from anonimous type that we have. It's workaround but I have doubts the problem
+        //Done: maybe resolved any different way:
+        return source.map((item:Site) => new Site(item.site, item.isMutated));
```

Comment:
Recreating of item(s) is bad approach. It means we must implement this ManuallyCast<T> interface everywhere where we need to use deep copy. Also what if there were several deep copies, as Site, Gene etc.? Then with this approach we must implement several interfaces: ManuallyCast<Site>, ManuallyCast<Gene>.